src/routes/api/v1/service-entries/+server.ts

```diff
@@ -0,0 +1,129 @@
+      #   service entries that live on core content pages (i.e service groups).
+      #
+      #  Be mindful of those details when monitoring the number of service entries returned
+      #  here vs the total number of entries that are published in Contentful.
+      #
+      #  The current number of service groups as of 10/31/2024 is 148. 200 gives us a
+      #  reasonable buffer while still staying under query complexity limits.
+      serviceGroupCollection(limit: 200, preview: false) {
+        total
+
+        items {
+          pageMetadata {
+            sys {
+              id
+            }
+          }
+          __typename
+          title
+
+          serviceEntriesCollection(limit: 50) {
+            items {
+              ... on ServiceEntry {
+                sys {
+                  id
+                }
+                __typename
+                entryTitle
+                description {
+                  json
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  `;
+
+  queryServiceEntries;
+
+  const data = await contentfulClient?.fetch<CoreContentwCtAsQuery>(
+    printQuery(queryServiceEntries),
+  );
+  type serviceEntries = ExtractQueryType<
+    CoreContentwCtAsQuery,
+    ["serviceGroupCollection", "items", "serviceEntryCollection", "items"]
+  >;
+
+  const serviceEntries: SearchIndexServiceEntry[] = [];
+  const duplicateEntries: Record<string, { count: number; parents: string[] }> = {};
+
+  processData: {
+    if (!data) break processData;
+
+    const { pageMetadataMap } = await loadPageMetadataMap({ contentfulClient });
+
+    data?.serviceGroupCollection?.items.forEach((serviceGroup) => {
+      serviceGroup?.serviceEntriesCollection?.items.forEach((serviceEntry) => {
+        if (
+          // appease TypeScript errors
+          serviceEntry &&
+          "entryTitle" in serviceEntry &&
+          serviceEntry.entryTitle &&
+          serviceGroup.title
+        ) {
+          if (duplicateEntries[serviceEntry?.entryTitle]) {
+            duplicateEntries[serviceEntry.entryTitle].count += 1;
+            duplicateEntries[serviceEntry.entryTitle].parents.push(serviceGroup.title);
+          } else {
+            duplicateEntries[serviceEntry.entryTitle] = {
+              count: 1,
+              parents: [serviceGroup.title],
+            };
+          }
+
+          if (serviceEntry?.description?.json) {
+            const parentPage = pageMetadataMap.get(serviceGroup?.pageMetadata?.sys?.id ?? "");
+            serviceEntries.push({
+              // objectID is the property needed to index content in Algolia,
+              objectID: serviceEntry.sys?.id,
+              url: `${parentPage?.url}#${slugify(serviceEntry.entryTitle)}`,
+              metaTitle: `${serviceGroup.title} | ${serviceEntry.entryTitle}`,
+              metaDescription: documentToPlainTextString(serviceEntry?.description?.json),
+              entryType: "Service Entry",
```

Comment:
The snippet logic is dependent on this value. I also like that this gives some extra properties to filter within Algolia if needed. It might be nice to have a shared constant variable instead of a string getting passed around though